sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -47,4 +48,69 @@ case class Md5(child: Expression)
       "org.apache.spark.unsafe.types.UTF8String.fromString" +
         s"(org.apache.commons.codec.digest.DigestUtils.md5Hex($c))")
   }
+
+}
+
+/**
+ * A function that calculates an SHA-1 digest and returns it as a hex string
+ * For input of type [[BinaryType]]
+ */
+case class Sha(child: Expression)
+  extends UnaryExpression with ExpectsInputTypes {
+
+  override def dataType: DataType = StringType
+
+  override def expectedChildTypes: Seq[DataType] = Seq(BinaryType, StringType)
```

Comment:
I am not sure. I already implemented the sha/sha1 (#6963). @Davies mentioned that `Seq(BinaryType, StringType)` is the way to go, because it should support `BinaryType` and `StringType`.